src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -85,21 +94,116 @@ public function setCallback($callback = null)
     }
 
     /**
+     * @param string $content
+     *
+     * @return Response
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function setContent($content)
+    {
+        if (!is_string($content)) {
+            throw new \InvalidArgumentException('Content is not a string. Use addData for adding data');
+        }
```

Comment:
I would remove this. The docblock already states that `$content` must be a string (we don't do similar checks anywhere else).